docs/release-notes/june-2023.md

```diff
@@ -0,0 +1,226 @@
+
+### Fixed
+
+* Taint analysis: Improve handling of dataflow for tainted value propagation in class field definitions
+This change resolves an issue where dataflow was not correctly accounted for when tainted values flowed through field definitions in class/object definitions. For instance, in Kotlin or Scala, singleton objects are commonly used to encapsulate executable logic, where each field definition behaves like a statement during object initialization. In order to handle this scenario, we have introduced an additional step to analyze a sequence of field definitions as a sequence of statements for taint analysis. This enhancement allows us to accurately track tainted values during object initialization. (gh-7742)
+* Allow any characters in file paths used to create dotted rule IDs. File path characters that aren't allowed in rule IDs are simply removed. For example, a rule whose ID is my-rule found in the file hello/@world/rules.yaml becomes hello.world.my-rule. (gh-8057)
+* Diff-aware scans now work when the git state isn't clean. (pa-2795)
+* Fixed a typing issue with Go. Previously, the pattern `($VAR : *tau.rho).$F()` wouldn't produce a match in the following:
+  ```
+  func f() {
+    i_1 := &tau.rho{}
+    i_2 := new(tau.rho)
+  
+    i_1.shift() //miss one
+    i_2.left()  //miss two
+  
+    return 101
+  }
+  ```
+but now Semgrep doesn't miss those two findings. (gh-6733)
+
+## Semgrep Cloud Platform
+
+### Added
+
+* You can now rename repositories.
+* You can now select a default RBAC (role-based access control) role.
+* The Policies page is now the default page used to manage rules.
+
+### Changed
+
+* For organizations with multiple GitHub organizations, you can now add organizations as a newline separated list.
+
+### Removed
+
+* The **Disliked** column has been removed from the Dashboard.
+
+## Semgrep Code
+
+### Added
+* Semgrep Pro engine taint-mode for Java: Semgrep can now relate Java properties and their corresponding getters/setters even when these are autogenerated, so the actual getters/setters are not declared in the sources. 
+* Semgrep Code findings page now includes security category tag.
+* Added a hide or show toggle for rules with no associated findings.
+* Users can now specify which rule boards to add rules to.
+* Enable user to select a range of cards at once while keeping Shift down.
+
+### Fixed
+* Cleaned up and updated wording and UI to reflect the updated Team tier. 
+
+## Semgrep Supply Chain
+
+* Semgrep Supply Chain now supports pNPm lockfiles (`pnpm-lock.yaml`) version 8 and above.
+* CLI scans now include the following metadata:
+    * Severity
+    * Recommended version to upgrade a dependency to
+
+## Semgrep Assistant (beta)
+
+* Add Settings toggle to enable users to choose where Semgrep Assistant findings appear. You can receive Semgrep Assistant messages within your Slack or GitHub PR comments.
```

Comment:
```suggestion
* Semgrep Assistant is now in public beta and can be enabled by anyone by installing a new GitHub app via the Settings page.
* A new setting allows you to configure where Semgrep Assistant's triage suggestions appear. You can receive Semgrep Assistant messages within your Slack or GitHub PR comments.
* A new setting allows setting a minimum confidence level required because autofix suggestions are posted by Assistant.
```